### PR TITLE
Change MGO to be Cluster-Scoped

### PR DESCRIPTION
**Pull request by dustman9000**
In order to ensure separation of concerns and ensure sensitive secrets and resources are inaccessible from CEE in the openshift-must-gather-operator namespace, we need to make sure MGO is cluster-scoped such that CRs can be read from any namespace (such as CEE's namespace).

* Manifests renamed to be consistent with operator-sdk v0.16.0 manifest generation file naming.
* Removed redundant ClusterRole and ClusterRoleBinding definitions.
* Updated OLM template (ran make).

Reference:
[v0.16.x Cluster Scope Guide](https://github.com/operator-framework/operator-sdk/blob/v0.16.x/doc/operator-scope.md#cluster-scoped-operator-usage)

### CODE REVIEW

File: deploy/03_role_binding.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 apiVersion: rbac.authorization.k8s.io/v1
```
Comment (2uasimojo) [inline deploy/03_role_binding.yaml:1]: Do we want to name this file something `cluster_role_binding`?

File: deploy/02_role.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-kind: Role
+kind: ClusterRole
```
Comment (2uasimojo) [inline deploy/02_role.yaml:1]: Do we want to name this file somtething `cluster_role`?

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 3]
```diff
@@ -27,38 +27,103 @@ objects:
         api.openshift.com/managed: 'true'
```
Comment (2uasimojo) [inline hack/olm-registry/olm-artifacts-template.yaml:27]: Validated this file properly generated from the above ✓

**Review by 2uasimojo — COMMENTED** [review]
This looks right to me. Minor quibble about the yaml filenames for the ClusterRole[Bindings].

This is latent, but I noticed the image is hardcoded to `:latest`. Are we planning to template that at some point and inject a real tag at build time, like we do for other operators?

**Comment by 2uasimojo** [comment]
Thanks for the updates.

/lgtm

**Comment by dustman9000** [comment]
> This is latent, but I noticed the image is hardcoded to `:latest`. Are we planning to template that at some point and inject a real tag at build time, like we do for other operators?

Yes, this will be taken care of in a subsequent PR.

**Pull request merged**